src/App.js

```diff
@@ -1,25 +1,99 @@
-import logo from './logo.svg';
 import './App.css';
+import React from 'react';
+import contacts from "./contacts.json";
+
+const five= [...contacts.slice(0, 5)];
+
+class Random extends React.Component {
+  state = {
+    contacts: five,
+  }
+
+  addNewRandom = (event) => {
+    let contactsCopy = [...this.state.contacts]
+    const random = contacts[Math.floor(Math.random() * contacts.length)];
+    if (contactsCopy.findIndex(el => el.id === random.id) !== -1) {
+      console.log(random)
+      return this.addNewRandom()
+    }
+    this.setState ({
+      contacts : [...this.state.contacts, {...random}]
```

Comment:
pas besoin de faire `{...random}`, il suffira de mettre `random` diretement comme : `[...this.state.contacts, random]`

Reply:
ok merci